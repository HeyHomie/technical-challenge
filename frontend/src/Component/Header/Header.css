header {
  background-color: #24292f;
}
.menu {
  display: flex;
  align-items: center;
  justify-content: flex-start;
  padding: 0px 50px;
  color: #fff;
  /* padding: 20px 50px; */
  /* box-shadow: 0 1px 8px #ddd; */
}

.menu-list {
  list-style: none;
  margin-bottom: 0;
}

.menu-list .list-item {
  display: inline-block;
  padding: 10px;
}

.list-item_margin {
  margin-right: 10px;
}

.list-item_btn {
  border: 1px solid #30363d;
  border-radius: 6px;
  padding: 4px 8px 4px 8px;
}

.search-input {
  background: inherit;
  border-radius: 6px;
  position: relative;
  padding: 5px 20px;
  font-size: 14px;
  border: 0;
  color: #fff;
}

.search-label {
  display: flex;
  align-items: center;
  background-color: inherit;
  border: 1px solid #30363d;
  border-radius: 6px;
}

.menu-list_link {
  text-decoration: none;
  color: #fff;
  transition: all 0.3s ease 0s;
}

.menu-list_link:hover {
  color: #c9d1d9;
}

.menu-right {
  display: flex;
  margin-left: auto;
  align-items: center;
}
