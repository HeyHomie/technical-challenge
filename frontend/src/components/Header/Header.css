.header {
  color: var(--color-header-text);
  background-color: var(--color-header-bg);
  padding-top: 8px;
  padding-bottom: 8px;
}

.header .mobile-nav {
  display: flex;
  justify-content: space-between;
  padding-right: 16px;
  padding-left: 16px;
  align-items: center;
}

.header .mobile-nav a {
  color: var(--color-scale-white);
}

.header .mobile-nav .auth-mobile {
  display: flex;
  align-items: center;
}

.header .auth-mobile button {
  background: transparent;
  border: 0;
  color: var(--color-scale-white);
  padding: 0;
}

.header .auth-mobile a, .header .auth-desktop .button {
  padding: 4px 8px;
  margin-right: 16px;
  display: inline-block;
  border-radius: 6px;
  border: 1px solid var(--color-neutral-muted);
}

.header .navigation, .header .auth-desktop {
  display: none;
}

@media (min-width: 1012px) {  

  .header {
    display: flex;
    padding: 8px 16px;
    align-items: center;
  }

  .header .mobile-nav {
    margin-right: 32px;
    padding: 0;
  }

  .header .mobile-nav .auth-mobile {
    display: none;
  }

  .header .navigation, .header .auth-desktop {
    display: flex;
  }

  .header .navigation {
    flex: 1;
  }

  .header .navigation ul {
    display: flex;
    margin: 0;
    padding: 0;
    list-style: none;
    align-items: center;
  }

  .header .navigation ul li {
    padding: 16px 0;
    margin-right: 16px !important;
  }

  .header .navigation ul li, .header .navigation ul li a {
    color: var(--color-scale-white);
  }

  .header .auth-desktop {
    display: flex;
    align-items: center;
  }

  .header .auth-desktop a {
    color: var(--color-scale-white);
  }

  .header .auth-desktop .button {
    margin-right: 0;
    margin-left: 16px;
  }
}