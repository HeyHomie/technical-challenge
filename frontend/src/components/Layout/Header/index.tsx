import React from 'react'
import './styles.css'

export const Header = (): JSX.Element => {
  return (
    <div className='Header--container'>
      <h2 className='Header__title'>GitHub</h2>
      <img
        className='Header__logo'
        src='https://cdn-icons-png.flaticon.com/512/25/25231.png'
        alt='Github logo'
      />
    </div>
  )
}
