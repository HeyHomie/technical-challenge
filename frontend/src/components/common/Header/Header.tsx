import {
  ThreeBarsIcon,
  MarkGithubIcon,
  BellIcon,
  PlusIcon,
  ChevronDownIcon,
  PersonIcon
} from '@primer/octicons-react'

const Header = () => (
  <header className="h-16 sm:h-[60px] p-4 bg-secondary text-sm">
    <div className="flex items-center justify-between sm:hidden">
      <ThreeBarsIcon size={24} />
      <MarkGithubIcon size={32} />
      <BellIcon size={24} />
    </div>
    <div className="items-center justify-between hidden sm:flex">
      <div className="flex items-center gap-4">
        <MarkGithubIcon size={32} />
        <input
          className="input max-w-[15.5rem]"
          placeholder="Search or jump to..."
        />
        <ul className="flex gap-4 font-bold">
          <li>
            <a href="https://github.com/pulls">Pulls</a>
          </li>
          <li>
            <a href="https://github.com/issues">Issues</a>
          </li>
          <li>
            <a href="https://github.com/marketplace">Marketplace</a>
          </li>
          <li>
            <a href="https://github.com/explore">Explore</a>
          </li>
        </ul>
      </div>
      <div className="flex gap-4">
        <BellIcon size={16} />
        <div className="flex items-center">
          <PlusIcon size={16} />
          <ChevronDownIcon size={10} className="my-auto" />
        </div>
        <div className="flex items-center">
          <PersonIcon size={20} className="border rounded-full" />
          <ChevronDownIcon size={10} />
        </div>
      </div>
    </div>
  </header>
)

export default Header
