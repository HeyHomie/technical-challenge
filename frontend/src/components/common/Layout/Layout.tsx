import { FC } from 'react'
import { Header } from 'components/common'

const Layout: FC = ({ children }) => {
  return (
    <div className="dark">
      <div className="w-screen min-h-screen text-primary bg-primary">
        <Header />
        {children}
      </div>
    </div>
  )
}

export default Layout
