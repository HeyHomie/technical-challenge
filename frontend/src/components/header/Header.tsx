import { FC } from 'react'

export const Header: FC = () => {
  return <header>all the main navigation is here</header>
}
