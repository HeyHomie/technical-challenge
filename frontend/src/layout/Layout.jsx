import React from 'react'
import { useGetData } from '../hooks/useGetData'

export const Layout = ({ children }) => {
  return (
    <>
      <header></header>
      <main>{children}</main>
      <footer></footer>
    </>
  )
}
