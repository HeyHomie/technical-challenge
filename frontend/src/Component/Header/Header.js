import React, { useEffect } from 'react'
import { Container, Row, Nav, Navbar, Col } from 'react-bootstrap'
import { withRouter, Link } from 'react-router-dom'
import Dropdown from '../Dropdown/Dropdown'
import dataFeatures from '../../help/dataFeatures.json'
import './Header.css'

export default function Header() {
  return (
    <header>
      <nav className="menu">
        <svg
          height="32"
          aria-hidden="true"
          viewBox="0 0 16 16"
          version="1.1"
          width="32"
          data-view-component="true"
          fill="white"
          className="octicon octicon-mark-github">
          <path
            fillRule="evenodd"
            d="M8 0C3.58 0 0 3.58 0 8c0 3.54 2.29 6.53 5.47 7.59.4.07.55-.17.55-.38 0-.19-.01-.82-.01-1.49-2.01.37-2.53-.49-2.69-.94-.09-.23-.48-.94-.82-1.13-.28-.15-.68-.52-.01-.53.63-.01 1.08.58 1.23.82.72 1.21 1.87.87 2.33.66.07-.52.28-.87.51-1.07-1.78-.2-3.64-.89-3.64-3.95 0-.87.31-1.59.82-2.15-.08-.2-.36-1.02.08-2.12 0 0 .67-.21 2.2.82.64-.18 1.32-.27 2-.27.68 0 1.36.09 2 .27 1.53-1.04 2.2-.82 2.2-.82.44 1.1.16 1.92.08 2.12.51.56.82 1.27.82 2.15 0 3.07-1.87 3.75-3.65 3.95.29.25.54.73.54 1.48 0 1.07-.01 1.93-.01 2.2 0 .21.15.46.55.38A8.013 8.013 0 0016 8c0-4.42-3.58-8-8-8z"></path>
        </svg>

        <ul className="menu-list">
          <li className="list-item">
            <Dropdown data={dataFeatures} name="Product" />
          </li>
          <li className="list-item">
            <Link to="/" className="menu-list_link">
              Team
            </Link>
          </li>
          <li className="list-item">
            <Link to="/" className="menu-list_link">
              Enterprise
            </Link>
          </li>
          <li className="list-item">
            <Link to="/" className="menu-list_link">
              Explore
            </Link>
          </li>
          <li className="list-item">
            <Link to="/" className="menu-list_link">
              Marketplace
            </Link>
          </li>
          <li className="list-item">
            <Link to="/" className="menu-list_link">
              Pricing
            </Link>
          </li>
        </ul>
        <div className="menu-right">
          <div className="list-item list-item_margin">
            <form className="search-input">
              <label htmlFor="" className="search-label">
                <input
                  type="text"
                  className="search-input"
                  placeholder="Search"
                />
                <svg
                  xmlns="http://www.w3.org/2000/svg"
                  width="22"
                  height="20"
                  aria-hidden="true"
                  className="mr-1 header-search-key-slash">
                  <path
                    fill="none"
                    stroke="#979A9C"
                    opacity=".4"
                    d="M3.5.5h12c1.7 0 3 1.3 3 3v13c0 1.7-1.3 3-3 3h-12c-1.7 0-3-1.3-3-3v-13c0-1.7 1.3-3 3-3z"></path>
                  <path fill="#979A9C" d="M11.8 6L8 15.1h-.9L10.8 6h1z"></path>
                </svg>
              </label>
            </form>
          </div>
          <div className="list-item list-item_margin">
            <Link to="/" className="menu-list_link">
              Sign In
            </Link>
          </div>
          <div className="list-item list-item_btn">
            <Link to="/" className="menu-list_link">
              Sign up
            </Link>
          </div>
        </div>
      </nav>
    </header>
  )
}
