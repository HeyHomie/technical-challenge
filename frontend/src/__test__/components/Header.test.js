import React from 'react'
import { render } from '@testing-library/react'
import { Header } from '../../components/Layout/Header'
import { create } from 'react-test-renderer'

describe('Header Component', () => {
  const header = render(<Header />)
  test('Render of the component Header', () => {
    expect(header).toBeInTheDocument
  })
})

describe('Footer Snapshot', () => {
  test('Comprobate UI from the component Header', () => {
    const header = create(
      <Header />
    )
    expect(header.toJSON()).toMatchSnapshot()
  })
})
