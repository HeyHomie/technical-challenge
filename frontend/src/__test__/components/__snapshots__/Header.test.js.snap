// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer Snapshot Comprobate UI from the component Header 1`] = `
<div
  className="Header--container"
>
  <h2
    className="Header__title"
  >
    GitHub
  </h2>
  <img
    alt="Github logo"
    className="Header__logo"
    src="https://cdn-icons-png.flaticon.com/512/25/25231.png"
  />
</div>
`;
